statistics/handle/ddl.go

```diff
@@ -44,21 +46,67 @@ func (h *Handle) HandleDDLEvent(t *util.Event) error {
 			}
 		}
 	case model.ActionAddTablePartition, model.ActionTruncateTablePartition:
-		pruneMode := h.CurrentPruneMode()
-		if pruneMode == variable.Static {
-			for _, def := range t.PartInfo.Definitions {
-				if err := h.insertTableStats2KV(t.TableInfo, def.ID); err != nil {
-					return err
-				}
+		for _, def := range t.PartInfo.Definitions {
+			if err := h.insertTableStats2KV(t.TableInfo, def.ID); err != nil {
+				return err
 			}
 		}
-		if pruneMode == variable.Dynamic {
-			// TODO: need trigger full analyze
-		}
 	case model.ActionDropTablePartition:
 		pruneMode := h.CurrentPruneMode()
 		if pruneMode == variable.Dynamic {
-			// TODO: need trigger full analyze
+			// We need to merge the partition-level stats to global-stats when we drop table partition in dynamic mode.
+			tableID := t.TableInfo.ID
+			is := infoschema.GetInfoSchema(h.mu.ctx)
+			h.mu.Lock()
+			globalTable, ok := h.getTableByPhysicalID(is, tableID)
+			h.mu.Unlock()
+			if !ok {
+				err := errors.Errorf("unknown physical ID %d in stats meta table, maybe it has been dropped", tableID)
+				return err
```

Comment:
What would happen if we return an error here? A warn log record?

Reply:
Yes, a warn log record will be generated.